library/entropy_poll.c

```diff
@@ -50,26 +50,42 @@
 
 #include <windows.h>
 #if _WIN32_WINNT >= 0x0501 /* _WIN32_WINNT_WINXP */
-#include <wincrypt.h>
+#include <bcrypt.h>
+#if defined(_MSC_VER) && _MSC_VER <= 1600
+/* Visual Studio 2010 and earlier issue a warning when both <stdint.h> and
+ * <intsafe.h> are included, as they redefine a number of <TYPE>_MAX constants.
+ * These constants are guaranteed to be the same, though, so we suppress the
+ * warning when including intsafe.h.
+ */
+#pragma warning( push )
+#pragma warning( disable : 4005 )
+#endif
+#include <intsafe.h>
+#if defined(_MSC_VER) && _MSC_VER <= 1600
```

Comment:
It's probably not necessary here, but to avoid repetition of pre-processor conditions like this (and the potential to get them out of sync when the file is updated in the future), we could `#define` something like `MBEDTLS_POP_WARNING_PRAGMA` when the warning is enabled and then use `defined(MBEDTLS_POP_WARNING_PRAGMA)` as the condition to do the `warning(pop)` - we do it this way in `aesce.c`